user-management/src/test/scala/com/normation/plugins/usermanagement/RoleComputationTest.scala

```diff
@@ -4,165 +4,80 @@ import com.normation.plugins.usermanagement.UserManagementService.computeRoleCov
 import com.normation.rudder.AuthorizationType
 import com.normation.rudder.Role
 import com.normation.rudder.Role.Custom
-import com.normation.rudder.RoleToRights
-import com.normation.rudder.RoleToRights.parseRole
+import com.normation.rudder.RudderRoles
+import com.normation.zio._
 import org.junit.runner.RunWith
 import org.specs2.mutable.Specification
 import org.specs2.runner.JUnitRunner
 
 @RunWith(classOf[JUnitRunner])
 class RoleComputationTest extends Specification {
+  def parseRoles(roles: List[String]) = RudderRoles.parseRoles(roles).runNow.toSet
+
   "Computation of Role Coverage over Rights" should {
 
     "return 'None' when parameters are empty" in {
-      val role = parseRole(Seq("user")).toSet
-      computeRoleCoverage(role, Set()) must beNone
-      computeRoleCoverage(Set(), Set(AuthorizationType.Compliance.Read)) must beNone
-      computeRoleCoverage(Set(), Set()) must beNone
+      (computeRoleCoverage(Set(Role.User), Set()) must beNone) and
+      (computeRoleCoverage(Set(), Set(AuthorizationType.Compliance.Read)) must beNone) and
+      (computeRoleCoverage(Set(), Set()) must beNone)
     }
 
     "return 'None' when authzs contains no_rights" in {
-      val role = parseRole(Seq("user")).toSet
-      computeRoleCoverage(role, Set(AuthorizationType.NoRights)) must beNone
-      computeRoleCoverage(role, Set(AuthorizationType.NoRights) ++ AuthorizationType.allKind) must beNone
+      (computeRoleCoverage(Set(Role.User), Set(AuthorizationType.NoRights)) must beNone) and
+      (computeRoleCoverage(Set(Role.User), Set(AuthorizationType.NoRights) ++ AuthorizationType.allKind) must beNone)
     }
 
     "return a 'Custom' role for empty intersection" in {
-      val role = parseRole(Seq("user")).toSet
-      computeRoleCoverage(role, Set(AuthorizationType.Compliance.Read)) match {
-        case Some(r) =>
-          r.head match {
-            case Custom(_) => success
-            case knowRole  => failure(s"Unexpected role : ${knowRole.name}")
-          }
-        case None    => success
-      }
+      computeRoleCoverage(Set(Role.User), Set(AuthorizationType.Compliance.Read)) must beEqualTo(
+        Some(Set(Role.forAuthz(AuthorizationType.Compliance.Read)))
+      )
```

Comment:
Test need to be kept as small as possible, else the intent is lost.  There is a lot of builtin comparators, use them